typing/types.ml

```diff
@@ -300,6 +300,12 @@ and constructor_tag =
   | Cstr_extension of Path.t * bool     (* Extension constructor
                                            true if a constant false if a block*)
 
+let equal_tag t1 t2 = match (t1, t2) with
+  | Cstr_constant i1, Cstr_constant i2 -> i2 = i1
+  | Cstr_block i1, Cstr_block i2 -> i2 = i1
+  | Cstr_extension (path1, b1), Cstr_extension (path2, b2) -> path1 = path2 && b1 = b2
+  | _ -> false
+
```

Comment:
I've addressed this and did not see any significant impact to performance. I suspect because the OCAML compiler detects that the pattern is exhaustive with code as it exists currently and therefore it does not need to actually perform the check once jumping to this case.